### PR TITLE
Support IE11

### PR DESCRIPTION
**Pull request by carvinlo**
Isolate the impact of the shell application on the global native objects of the pilet and reduce the bundle size of the shell application. Use useBuiltIns to support IE11.

### CODE REVIEW

File: packages/piral-cli-webpack/src/configs/common.ts [hunk 1]
```diff
@@ -64,7 +64,22 @@ export function getStyleLoader() {
 
 export function getRules(baseDir: string): Array<RuleSetRule> {
   const styleLoader = getStyleLoader();
-  const nodeModules = resolve(baseDir, 'node_modules');
+
+  const PATH_DELIMITER = '[\\\\/]'; // match 2 antislashes or one slash
```
Comment (FlorianRappl) [inline packages/piral-cli-webpack/src/configs/common.ts]: Naming -> `pathDelimiter`.

File: packages/piral-cli-webpack/src/configs/common.ts [hunk 2]
```diff
@@ -87,28 +102,53 @@ export function getRules(baseDir: string): Array<RuleSetRule> {
...
+              '@babel/plugin-transform-modules-commonjs',
+              ['@babel/plugin-proposal-decorators', { legacy: true }],
+              ['@babel/plugin-proposal-class-properties', { loose: true }],
+            ],
           },
         },
       ],
       exclude: nodeModules,
     },
-    {
```
Comment (FlorianRappl) [inline packages/piral-cli-webpack/src/configs/common.ts]: Removing this I would then add the [fork-ts-checker-webpack-plugin](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin).

File: packages/piral-cli-webpack/src/configs/common.ts [hunk 3]
```diff
@@ -87,28 +102,53 @@ export function getRules(baseDir: string): Array<RuleSetRule> {
...
       use: [
         {
           loader: 'babel-loader',
           options: {
-            presets: ['@babel/preset-env', '@babel/preset-react'],
+            presets: [
+              [
+                '@babel/preset-env',
+                {
+                  targets: {
```
Comment (FlorianRappl) [inline packages/piral-cli-webpack/src/configs/common.ts:119]: This should be all given by the end user.

By default it should just use `preset-env`. If users change, e.g., their package.json then this should be reflected.

We should not force IE9/10 to *any* user and it should be easy to opt-out of IE 11.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
Looks good so far, but still misses CHANGELOG and package.json / version changes.

File: packages/piral-cli-webpack/src/configs/common.ts (hunk 2 above)
Comment (carvinlo) [reply packages/piral-cli-webpack/src/configs/common.ts]: Do you mean to remove `@babel/preset-typescript`?
I haven't figured out why the ts-loader did not convert the arrow function of the component.

File: packages/piral-cli-webpack/src/configs/common.ts (hunk 3 above)
Comment (carvinlo) [reply packages/piral-cli-webpack/src/configs/common.ts:119]: You are right, moving the configuration to the package.json of the shell application is a more flexible and free configuration method.
I strongly agree to remove the support of ie <11 and there are still some users who have a demand for ie11.

File: packages/piral-cli-webpack/src/configs/common.ts (hunk 1 above)
Comment (carvinlo) [reply packages/piral-cli-webpack/src/configs/common.ts]: my mistake.

File: packages/piral-cli-webpack/src/configs/common.ts (hunk 2 above)
Comment (FlorianRappl) [reply packages/piral-cli-webpack/src/configs/common.ts]: I think its either ts-loader or babel with ts + the fork-ts-checker. The latter combo could be the best in terms of performance and flexibility.

Regarding the arrow function: Is it maybe due to the used tsconfig? Maybe the target there is ES6?

Comment (carvinlo) [reply packages/piral-cli-webpack/src/configs/common.ts]: Aha, you may be right, I mistaken the `tsconfig.json` in the `piral-cli-webpack` directory for the `tsconfig.json` in the shell application directory, and I will come back and resubmit it after verifying no problem.

Comment (carvinlo) [reply packages/piral-cli-webpack/src/configs/common.ts]: The problem has been resolved and verified

**Review by FlorianRappl — APPROVED** [review]
LGTM!

**Comment by FlorianRappl** [comment]
(Still no version / CHANGELOG has been updated, so this will fail publishing ...)

**Comment by carvinlo** [comment]
> (Still no version / CHANGELOG has been updated, so this will fail publishing ...)

After updating change.log, it still fail publishing. Is there anything else I need to do?

**Review by FlorianRappl — COMMENTED** [review]
LGTM!

**Pull request merged**